File: bank-service/src/main/java/rs/edu/raf/model/credit/CreditType.java
package rs.edu.raf.model.credit;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import rs.edu.raf.model.Currency;

import java.math.BigDecimal;

@Entity
@Getter
@Setter
public class CreditType {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private BigDecimal nominalInterestRate;
    private int minLoanTerm;
    private int maxLoanTerm;
    private BigDecimal maxLoanAmount;
    private BigDecimal prepayment;
    @ManyToOne
    private Currency currency;
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/model/credit/CreditType.kt
package org.example.bankservicekotlin.model.credit

import jakarta.persistence.*
import org.example.bankservicekotlin.model.Currency
import java.math.BigDecimal

@Entity
class CreditType(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    val name: String,
    val nominalInterestRate: BigDecimal,
    val minLoanTerm: Int,
    val maxLoanTerm: Int,
    val maxLoanAmount: BigDecimal,
    val prepayment: BigDecimal,
    @ManyToOne
    val currency: Currency
) {
}